### PR TITLE
Clean up splash attention imports

### PR DESCRIPTION
**Pull request by huytransformer**
# Description

The general splash attention path now imports from the `tokamax` dependency while the ring attention keeps the adapted kernels in `kernels/tokamax_splash_attention` via `tokamax_ring_attention`. 

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/layers/attention_op.py [hunk 1]
```diff
@@ -77,6 +75,8 @@
 import numpy as np
 from tokamax._src.ops.attention import base as tokamax_attention_base
 from tokamax._src.ops.attention import pallas_triton as tokamax_pallas_triton
+from tokamax._src.ops.experimental.tpu.splash_attention import splash_attention_kernel as tokamax_splash_kernel
+from tokamax._src.ops.experimental.tpu.splash_attention import splash_attention_mask as tokamax_splash_mask
```
Comment (NuojCheng) [inline src/maxtext/layers/attention_op.py:79]: I would expect an if-else path, e.g. if using ring attention we take forked splash otherwise tokamax path. If the changes are simple like this, why we need to fork all files from tokamax at the beginning?

Comment (huytransformer) [reply src/maxtext/layers/attention_op.py:79]: Ring never needed this, it uses the `tokamax_ring_attention`. These were originally intended for consistency of splash paths

**Review by NuojCheng — APPROVED** [review]
